xbmc/pvr/channels/PVRChannelGroup.cpp

```diff
@@ -498,9 +498,10 @@ int CPVRChannelGroup::GetMembers(CFileItemList &results, bool bGroupMembers /* =
 
   for (PVR_CHANNEL_GROUP_SORTED_MEMBERS::const_iterator it = channels->m_sortedMembers.begin(); it != channels->m_sortedMembers.end(); ++it)
   {
-    if (bGroupMembers || !IsGroupMember((*it).channel))
+    const CPVRChannelPtr channel = (*it).channel;
+    if (bGroupMembers != channel->IsHidden() || !IsGroupMember(channel))
```

Comment:
This logic is a bit hard to understand, a longer more expressive boolean statement would help

Reply:
Agreed, took me a while to understand this as well. Will see how I can improve it to get more readable.